### PR TITLE
Don't use insecure temporary directory as home directory

### LINKED ISSUE #6501: tmp_home_path insecure
Hi,

If the home directory of the user is not writable the function tmp_home_path from lib/bundler.rb, creates an insecure temporary directory in tmp/:
- it is a world writable
- it has a with a predictable name.
Moreover, this temporary directory is not removed at the end.

See:
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=796383
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=881749
for more details.

Thanks in advance.

### PR DESCRIPTION
**Pull request by fatkodima**
Closes #6501

### CODE REVIEW

File: lib/bundler.rb [hunk 1]
```diff
@@ -177,16 +176,12 @@ def user_home
       end
     end
 
-    def tmp_home_path(login, warning)
-      login ||= "unknown"
+    def tmp_home_path(_login, warning)
```
Comment (deivid-rodriguez) [inline lib/bundler.rb]: I assume you didn't change the signature to remove the `login` parameter for backwards compatibility?

I don't think anybody should be using this method, and it's returning an insecure path anyways, so I would vote for making it private and removing the now unused parameter.

Thoughts?

File: spec/bundler/bundler_spec.rb [hunk 2]
```diff
@@ -282,31 +276,17 @@
...
-  describe "#tmp_home_path" do
-    it "should create temporary user home" do
-      allow(Dir).to receive(:tmpdir).and_return("/TMP")
-      allow(FileTest).to receive(:exist?).with("/TMP/bundler/home").and_return(false)
-      expect(FileUtils).to receive(:mkpath).once.ordered.with("/TMP/bundler/home")
-      expect(FileUtils).to receive(:mkpath).once.ordered.with("/TMP/bundler/home/USER")
-      expect(File).to receive(:chmod).with(0o777, "/TMP/bundler/home")
-      expect(Bundler.tmp_home_path("USER", "")).to eq(Pathname("/TMP/bundler/home/USER"))
-    end
-  end
```
Comment (deivid-rodriguez) [inline spec/bundler/bundler_spec.rb]: I agree with removing this test if we decide to make this method private.

We could consider a graceful deprecation but I really doubt anybody is relying on the result of this method for anything?

**Comment by fatkodima** [comment]
@deivid-rodriguez Updated with your suggestion of making method private. I am doubt too that anybody uses it.

**Comment by deivid-rodriguez** [comment]
@bundlerbot merge

**Pull request merged**